submodules/strategyHours/strategyHours.js

```diff
@@ -197,9 +197,13 @@ define(function(require) {
 				event.preventDefault();
 
 				monster.pub('voip.strategy.addOfficeHours', {
-					existing: self.strategyHoursGetDaysIntervalsFromTemplate(parent),
-					callback: function(err, existing) {
-						var normalizedIntervals = self.strategyHoursNormalizeDaysIntervals(existing);
+					callback: function(err, selected) {
+						var existing = self.strategyHoursGetDaysIntervalsFromTemplate(parent);
+							normalizedIntervals = _
+								.chain(existing)
+								.zipWith(selected, _.concat)
+								.thru(_.bind(self.strategyHoursNormalizeDaysIntervals, self))
+								.value();
```

Comment:
That works, although it might not be the most readable.

Chaining is usually more readable when the original intput does not change much, or when it's not a complex structure (like nested arrays), as it makes it harder to hold in your head the current value being work on.

To improve on that, we can used what's called a "bucket brigade"-style variable declarations, in which each function produces a result that the next function needs, so there is no reasonable way to call them out of order.

```suggestion
						var existing = self.strategyHoursGetDaysIntervalsFromTemplate(parent);
						      merged = _.zipWith(existing, selected, _.concat),
						      normalized = self.startegyHoursNormalizeDaysIntervals(merged);
```

Deciding which construct to use (chaining vs "bucket brigade") is at the discretion of the developer and depends on the complexity of the operations being performed.

Feel free to implement this, or not, if you don't feel like it improves readability 😉

Reply:
Sure, that makes sense.  I'll make the change suggested.